core/coreapi/interface/options/unixfs.go

```diff
@@ -0,0 +1,187 @@
+package options
+
+import (
+	"errors"
+	"fmt"
+
+	cid "gx/ipfs/QmPSQnBKM9g7BaUcZCvswUJVscQ1ipjmwxN5PXCjkp9EQ7/go-cid"
+	mh "gx/ipfs/QmPnFwZ2JXKnXgMw8CdBPxn7FWh6LLdjUjxV1fKHuJnkr8/go-multihash"
+	dag "gx/ipfs/QmcBoNcAP6qDjgRBew7yjvCqHq7p5jMstE44jPUBWBxzsV/go-merkledag"
+)
+
+type Layout int
+
+const (
+	BalancedLayout Layout = iota
+	TrickleLayout
+)
+
+type UnixfsAddSettings struct {
+	CidVersion int
+	MhType     uint64
+
+	InlineLimit  int
```

Comment:
A size 0 as an inline limit is useful.  You should have a separate option for enabling inlining and one for setting the limit.

Reply:
It is also far more user friendly to just be able to enable inlining and have it use the default value,